resources/abi/README.md

```diff
@@ -0,0 +1,74 @@
+# ABI change monitoring
+
+When the OpenAssetIO C++ ABI changes, we need to know about it to inform
+our release notes and semver versioning.
+
+This is aided (though not solved!) by automated testing against a
+snapshot dump of the ABI. The tool we use for this is [libabigail](https://sourceware.org/libabigail/manual/libabigail-overview.html),
+in particular its `abidiff` and `abidw` tools.
+
+> **Note**
+>
+> The ABI on other platforms, including other Linux distributions, may
+> (subtly) vary. Always generate a dump of a build from the same Docker
+> image as is used on CI, i.e. `ghcr.io/openassetio/openassetio-build`.
+
+If the ABI of an OpenAssetIO library changes, and the change is
+unavoidable, then a new XML snapshot dump of the ABI must be created
+(and release notes updated as appropriate).
+
+To do this we use `abidw`. For convenience, the `openassetio-build`
+Docker image has libabigail pre-installed.
+
+The generated snapshot is then checked against new builds using
+`abidiff`.
+
+## Example of updating the ABI snapshot
```

Comment:
[Consider] Great walkthrough, but since it's a neccesary workflow, I wonder if a link from PROCESS.md might be desireable

Reply:
Added a few links and mentions in 7450e2ae

Reply:
:+1: